compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUOps.td

```diff
@@ -303,38 +320,31 @@ def IREEGPU_CoalescedGatherDMAOp : Op<IREEGPU_Dialect, "coalesced_gather_dma", [
 
     1. Outer `scf.forall` represents workgroup-level parallelism.
     2. Inner `scf.forall` represents subgroup-level parallelism.
-    3. Each thread in the subgroup coalesces its memory accesses when
+    3. Each thread in the subgroup coalesces its memory accesses a slice of tensor
        gathering data and writes to its lane-offset in the destination tensor.
 
+    Here is an example of using this op in tiled situation, copy mode:
     ```mlir
-    %result = scf.forall (%wg_i, %wg_j) in (16, 1) shared_outs(%wg_out = %dest) -> (tensor<128x16xf32>) {
-      %indices_wg_slice = tensor.extract_slice ...
-      %dest_wg_slice = tensor.extract_slice ...
-
-      %inner_result = scf.forall (%sg_i, %sg_j) in (32, 1) shared_outs(%sg_out = %dest_wg_slice) -> (tensor<8x16xf32>) {
-        scf.forall.in_parallel {
-          iree_gpu.coalesced_gather_dma %indices_wg_slice, %source into %sg_out : ...
-        }
-      } {mapping = [#gpu.thread<linear_dim_1>, #gpu.thread<linear_dim_0>]}
-
+    %result = scf.forall (%arg5, %arg6) = (0, 0) to (1, 32) step (1, 32) shared_outs(%arg7 = %extracted_slice) -> (tensor<1x128xf32>) {
+      %extracted_slice_1 = tensor.extract_slice %extracted_slice[%arg5, %arg6] ... // extract thread slice
       scf.forall.in_parallel {
-        tensor.parallel_insert_slice %inner_result into %wg_out[...] : ...
+        %8 = iree_gpu.coalesced_gather_dma %extracted_slice_1 into %arg7 : tensor<1x4xf32>, tensor<1x128xf32> -> tensor<1x128xf32>
       }
-    } {mapping = [#gpu.warp<linear_dim_1>, #gpu.warp<linear_dim_0>]}
+    } {mapping = [#gpu.thread<linear_dim_1>, #gpu.thread<linear_dim_0>]}
     ```
   }];
 
   let arguments = (ins
-    AnyTypeOf<[RankedTensorOf<[Index]>, MemRefOf<[Index]>]>:$indices,
     AnyRankedTensorOrMemRef:$source,
+    Optional<VectorOfAnyRankOf<[Index]>>:$indices,
     AnyRankedTensorOrMemRef:$init
   );
 
   let results = (outs AnyRankedTensorOrMemRef:$result);
```

Comment:
Nit since I'm here ... bufferized ops don't have a `result`. So this should be an optional tensor, no?

Reply:
made `result` go away after bufferization.